src/main/java/com/github/hokkaydo/eplbot/module/code/JavaRunner.java

```diff
@@ -0,0 +1,146 @@
+        if (!input.equals(safeImports(input))){
+            return "Invalid imports";
+        }
+        if (requiresWrapper(input)){
+            input = addWrapper(input);
+        }
+        String className = regexClassName(input);
+        writeFile(input,Path.of(OUTPUT_PATH+File.separator+className+".java"));
+        String filePath = OUTPUT_PATH + File.separator + className + ".java";
+
+        ByteArrayOutputStream errorStream = new ByteArrayOutputStream();
+        if (ToolProvider.getSystemJavaCompiler().run(null, null, new PrintStream(errorStream), filePath) != 0) {
+            deleteFiles( className);
+            return "Compilation failed:\n" + errorStream;
+        }
+
+        ScheduledFuture<?> timeOut = SCHEDULER.schedule(() -> {}, runTimeout, TimeUnit.SECONDS);
+
+        ByteArrayOutputStream outputStream = new ByteArrayOutputStream();
+        try {
+            Process process = new ProcessBuilder(System.getProperty("java.home") + File.separator + "bin" + File.separator + "java", "-cp", System.getProperty("java.class.path") + File.pathSeparator + OUTPUT_PATH, regexClassName(input)).redirectErrorStream(true).start();
+            BufferedReader reader = new BufferedReader(new InputStreamReader(process.getInputStream()));
+            String line;
+            while ((line = reader.readLine()) != null) {
+                outputStream.write((line + "\n").getBytes());
+            }
+
+        } catch (Exception e) {
+            e.printStackTrace();
+            return "Server side error code J01" + e.getMessage();
+        }
+        if (timeOut.isDone()) {
+            deleteFiles( className);
+            String output = outputStream.toString().trim();
+            if (output.isEmpty()) {
+                return "Run failed: Timelimit exceeded "+ runTimeout +" s";
+            } else {
+                return "Run failed:\n" + output;
+            }
+        }
+        timeOut.cancel(true);
+        deleteFiles( className);
+        return outputStream.toString();
+            
+
+
+
+    }
+    
+    public static void deleteFiles(String className) {
+        Optional.ofNullable(new File(OUTPUT_PATH).listFiles())
+                .stream()
+                .flatMap(Arrays::stream)
+                .filter(f -> f.isFile() && f.getName().startsWith(className))
+                .forEach(File::delete);
+    }
+    public static boolean requiresWrapper(String javaCode) {
+        boolean hasClass = Pattern.compile("\\bpublic\\s+class\\s+[A-Z][a-zA-Z0-9]*").matcher(javaCode).find();
+        boolean hasMainMethod = Pattern.compile("\\bpublic\\s+static\\s+void\\s+main\\s*\\(\\s*String\\[\\]\\s+[a-zA-Z0-9]*\\s*\\)").matcher(javaCode).find();
+        boolean hasImports = Pattern.compile("\\bimport\\s+[a-zA-Z0-9.]+").matcher(javaCode).find();
+        if (hasClass && hasMainMethod) {
+            return false;
+        } else if (hasImports) {
+            return false;
+        } else {
+            return true;
```

Comment:
We require a wrapper only if hasClass and hasMainMethod are false. 
```suggestion
    return !hasMainMethod && !hasClass;
```